cas-server-support-rest/src/main/java/org/jasig/cas/support/rest/RegisteredServiceResource.java

```diff
@@ -0,0 +1,178 @@
+import org.jasig.cas.ticket.TicketGrantingTicket;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * {@link org.springframework.web.bind.annotation.RestController} implementation of a REST API
+ * that allows for registration of CAS services. Services will automatically be put in in the service registry,
+ * but are always disabled by default until they are approved by the explicit permission of the CAS deployer.
+ * @author Misagh Moayyed
+ * @since 4.2
+ */
+@RestController()
+public class RegisteredServiceResource {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegisteredServiceResource.class);
+
+    @Autowired
+    @Qualifier("servicesManager")
+    private ServicesManager servicesManager;
+
+    @Autowired
+    @Qualifier("centralAuthenticationService")
+    private CentralAuthenticationService centralAuthenticationService;
+
+    @Value("${cas.rest.services.attributename:}")
+    private String attributeName;
+
+    @Value("${cas.rest.services.attributevalue:}")
+    private String attributeValue;
+
+    /**
+     * Create new service.
+     *
+     * @param requestBody service application/x-www-form-urlencoded value
+     * @param tgtId ticket granting ticket id URI path param
+     * @return {@link ResponseEntity} representing RESTful response
+     */
+    @RequestMapping(value = "/services/add/{tgtId:.+}", method = RequestMethod.POST, consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE)
+    public final ResponseEntity<String> createService(@RequestBody final MultiValueMap<String, String> requestBody,
+                                                      @PathVariable("tgtId") final String tgtId) {
+        try {
+
+            if (StringUtils.isBlank(this.attributeName) || StringUtils.isBlank(this.attributeValue)) {
+                throw new IllegalArgumentException("Attribute name and/or value must be configured");
+            }
+
+            final TicketGrantingTicket ticket =
+                this.centralAuthenticationService.getTicket(tgtId, TicketGrantingTicket.class);
+            if (ticket == null || ticket.isExpired()) {
```

Comment:
So the only thing I need to add a new service is to be authenticated: I was expecting some Spring security involved here, but maybe it's overkill.